Rev0.95 rev  - Part spec folder reorganization

asadate opened the pull request:
Reorganization of the part-spec folder into multiple folders with standalone specs for each component.

asadate committed:
Create resistor.json

Move resistor specs out of passives.json

asadate committed:
Create unit.json

Move unit object specification to the common folder

asadate committed:
Update and rename part-spec/ratio.json to part-spec/common/ratio.json

Move ration object specification to common folder

asadate committed:
Delete part-spec/unit.json

delete unit object specification from folder.  This is now under the common folder

asadate committed:
Update and rename part-spec/package.json to part-spec/common/package.json

Moved package object definition to common folder

asadate committed:
Update and rename part-spec/externalFile.json to part-spec/common/externalFile.json

asadate committed:
Update and rename registers.json to register.json and move to common folder

asadate committed:
Update and rename part-spec/register.json to part-spec/common/register.json

asadate committed:
Update and rename part-spec/graph.json to part-spec/common/graph.json

asadate committed:
Update and rename part-spec/currentConsumption.json to part-spec/common/currentConsumption.json

asadate committed:
Update and rename part-spec/pinPaths.json to part-spec/common/pinPaths.json

asadate committed:
Update and rename part-spec/conditionalProperty.json to part-spec/common/conditionalProperty.json

asadate committed:
Update and rename part-spec/componentProtectionThresholds.json to part-spec/common/componentProtectionThresholds.json

Moved object specifications to common folder. Fixed camel notation on properties & added Hysteresis property OVP & UVLO

asadate committed:
Update pinPaths.json

Added missing title.

asadate committed:
Update and rename part-spec/pinSpec.json to part-spec/common/pinSpec.json

Added previously defined pinPaths object to pinSpec.

asadate committed:
Update and rename part-spec/componentID.json to part-spec/common/componentID.json

asadate committed:
Update component.json

Changed references to point to the updated folder.

asadate committed:
Create capacitor.json

Move capacitor specifications from passives.json from.

asadate committed:
Create inductor.json

Moved inductor specifications from passives,json file

asadate committed:
Create ferrite_bead.json

Moved ferrite bead specifications from passives.json file

asadate committed:
Create common_mode_choke.json

Moved common mode choke specifications from passives.json

asadate committed:
Delete part-spec/passives.json

Split components specifced in passives.json into multiple files.

asadate committed:
Update and rename part-spec/clock_oscillators.json to part-spec/clock/oscillator.json

Moved oscillator specifications to new clock folder

asadate committed:
Create clock.json

Moved clock specifications out of ic_logic.json

asadate committed:
Update and rename part-spec/ic_logic.json to part-spec/logic/logic_gate.json

Moved logic gate specifications to new folder

asadate committed:
Update and rename part-spec/adc_dac.json to part-spec/data_converter/adc.json

Move adc specifications to data_converter folder

asadate committed:
Create dac.json

Moved dac specifications to data_conveter folder.

asadate committed:
Update and rename part-spec/ic_micro.json to part-spec/ic_microcontroller/microcontroller.json

Moved microcontroller specifications to new folder

asadate committed:
Update and rename part-spec/ic_misc.json to part-spec/ic_misc/speaker_amplifier.json

Moved speaker amplifier specifications from ic_misc.json to new folder.

asadate committed:
Create audio_codec.json

Moved audio codec specifications out of ic_misc.json to new folder

asadate committed:
Create wlan_module.json

Moved wlan module specifications to new folder

asadate committed:
Create wwan_module.json

Moved wwan module specifications from ic_misc.json to new folder

asadate committed:
Create tpm.json

Moved tpm specifications from ic_misc.json to new folder

asadate committed:
Update and rename part-spec/usbc.json to part-spec/ic_misc/pd_controller.json

Move usbc pd controller specifications to new folder

asadate committed:
Update and rename part-spec/ic_io.json to part-spec/ic_io/redriver.json

Move redriver specifications from ic_io.json to new folder

asadate committed:
Create retimer.json

Moved retimer specifications out of ic_io.json to new folder

asadate committed:
Create bridge_chip.json

Moved bridge chip specifications out of ic_io.json to new folder

asadate committed:
Create level_shifter

Moved level shifter specifications to new folder.

asadate committed:
Create highspeed_mux.com

Moved high speed mux definition to new folder

asadate committed:
Update and rename pd_controller.json to usbc_portcontroller.json

asadate committed:
Update usbc_portcontroller.json

asadate committed:
Update and rename usbc_portcontroller.json to usbc_portcontroller.json

asadate committed:
Update and rename usbc_portcontroller.json to usbc_pdcontroller.json

asadate committed:
Create usbc_portcontroller.json

Added specifications for the USB Type-c port controller.

asadate committed:
Create usb_bc12

Added specifications for battery charger 1.2 detection

asadate committed:
Create switch.json

Moved switch definition to new folder

asadate committed:
Update and rename part-spec/hardware.json to part-spec/hardware/connector.json

Moved connector specifications to new folder

asadate committed:
Update and rename part-spec/led.json to part-spec/semiconductor/led.json

Moved Led specifications to new folder

asadate committed:
Create mosfet.json

Moved mosfet specifications to new folder

asadate committed:
Create diode.json

Moved diode specification to new folder

asadate committed:
Update and rename part-spec/semiconductor.json to part-spec/semiconductor/bjt.json

Moved bjt specifications to new folder. Fixed some units and added missing properties

asadate committed:
Create ssd.json

Moved ssd specifications to new folder.

asadate committed:
Create sd_card.json

Moved sd card specification to new folder

asadate committed:
Create dram.json

Added timing properties and moved dram specifications to new folder

asadate committed:
Create rom.json

Moved rom specifications to new folder

asadate committed:
Update dram.json

asadate committed:
Create eeprom

Moved eeprom specifications to new folder

asadate committed:
Update and rename part-spec/storage_memory.json to part-spec/memory/flash_memory.json

Moved flash memory specifications moved to new folder

asadate committed:
Create accelerometer.json

Moved accelerometer specifications to new folder

asadate committed:
Create gyroscope.json

Moved gyroscope specifications to new folder

asadate committed:
Create magnetic_sensor.json

Moved magnetic sensor specifications to new folder

asadate committed:
Update and rename part-spec/sensor.json to part-spec/sensor/thermal_sensor.json

Moved thermal sensor specifications to new folder

asadate committed:
Create powerFetProperties.json

Moved power FET properties to new folder

asadate committed:
Create switching_regulator.json

Moved switching regulator specifications to new folder.

asadate committed:
Create linear_regulator.json

Moved linear regulator specifications to new folder

asadate committed:
Create load_switch.json

Moved load switch specifications to new folder.

asadate committed:
Create displaybacklight_driver.json

Moved display backlight driver specifications to new folder

asadate committed:
Create battery_charger.json

Moved battery charger specifications to new folder

asadate committed:
Update and rename part-spec/power.json to part-spec/power/pmic.json

Moved pmic specifications to new folder. Updated properties for more clarity.

asadate committed:
Update pinSpec.json

Included proposed changes ( from Temitope). Removed some min/max properties to align with updated unit object specifications.

rsnancollas commented on part-spec/clock/clock.json:
```diff
@@ -0,0 +1,54 @@
+            "description": "number of clock outputs in a clock IC",
+            "type": "number"
+        },
+        "diffSingleEnded": {
```
differentialSingleEnded

asadate replied:
done

rsnancollas commented on part-spec/clock/clock.json:
```diff
@@ -0,0 +1,54 @@
+        "diffSingleEnded": {
+            "description": "property describing whether a clock output is single ended or differential",
+            "enum": [
+                "diff",
```
differential

asadate replied:
done

rsnancollas commented on part-spec/clock/clock.json:
```diff
@@ -0,0 +1,54 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "frequencyTolerance": {
+            "description": "amount of frequency variation specced from nominal frequency",
```
"amount of frequency variation from nominal frequency"

asadate replied:
done

rsnancollas commented on part-spec/clock/clock.json:
```diff
@@ -0,0 +1,54 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "powerSupplyRejectionRatio": {
+            "description": "power supply rejection ratio (PSRR)or ratio between power supply variation and output variation",
```
power supply rejection ratio (PSRR) or ratio between power supply variation and output variation

asadate replied:
done

rsnancollas commented on part-spec/clock/oscillator.json:
```diff
@@ -0,0 +1,65 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "frequencyStability": {
+            "description": "Frequency change over temperature, load, supply voltage change and aging",
```
Frequency -> frequency

asadate replied:
done

rsnancollas commented on part-spec/clock/oscillator.json:
```diff
@@ -0,0 +1,65 @@
+        },
+        "frequencyStability": {
+            "description": "Frequency change over temperature, load, supply voltage change and aging",
+            "$comment": "units of ppm",
```
this unit doesn't exist in "unit" - should we add it or just have this be a number?

asadate replied:
Changed to number.

rsnancollas commented on part-spec/clock/clock.json:
```diff
@@ -0,0 +1,54 @@
+        },
+        "powerSupplyRejectionRatio": {
+            "description": "power supply rejection ratio (PSRR)or ratio between power supply variation and output variation",
+            "comment": "units of dB",
```
this unit doesn't exist in "unit" - should we add it or just have this be a number?

asadate replied:
Let's have this as a number. Changed

rsnancollas commented on part-spec/common/componentProtectionThresholds.json:
```diff
@@ -34,28 +34,38 @@
-                "undervoltageLockoutThresholdFalling": {
-                    "description": "Undervoltage Lockout Out (UVLO) Threshold with monitored supply Falling",
+                "underVoltageLockoutThresholdRising": {
+                    "description": "Undervoltage Lockout Out (UVLO) Threshold with power supply rising",
```
remove the extra "Out"

asadate replied:
done

rsnancollas commented on part-spec/common/componentProtectionThresholds.json:
```diff
@@ -34,28 +34,38 @@
+                    "$ref": "../common/unit.json#/unit"
+                },
+                "underVoltageLockoutThresholdFalling": {
+                    "description": "Undervoltage Lockout Out (UVLO) Threshold with power supply Falling",
```
remove the extra "Out"

rsnancollas replied:
Falling -> falling

asadate replied:
done

rsnancollas commented on part-spec/common/componentProtectionThresholds.json:
```diff
@@ -34,28 +34,38 @@
+                    "$ref": "../common/unit.json#/unit"
+                },
+                "underVoltageLockoutHysteresis": {
+                    "description": "Undervoltage Lockout Out (UVLO) Hysteresis",
```
remove the extra "Out"

asadate replied:
done

rsnancollas commented on part-spec/common/componentProtectionThresholds.json:
```diff
@@ -34,28 +34,38 @@
             "properties": {
                 "supplyName": {
-                    "description": "Name of the supply monitored",
+                    "description": "Name of the power supply",
```
Name -> name

asadate replied:
Changed

rsnancollas commented on part-spec/common/pinPaths.json:
```diff
@@ -1,6 +1,7 @@
-  "$id": "https://github.com/edatasheets/edatasheets.github.io/blob/main/part-spec/pinPaths.json",
+  "$id": "https://github.com/edatasheets/digital-datasheets/blob/main/part-spec/common/pinPaths.json",
   "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": " specifications of pinPaths", 
```
Specification of pin paths through an IC

asadate replied:
done

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+{
+    "$id": "https://github.com/edatasheets/digital-datasheets/blob/main/part-spec/common/pinSpec.json",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "specifications of pins",
```
specifications -> Specification

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                "description": "description of the signal appearing at the terminal of an electric/electronic component",
+                "type": "string"
+            },
+            "numberOfSupportedFunctions": {
```
still not sure if we should have this since it overlaps with functionProperties

asadate replied:
Since this is optional, I believe we can keep it.

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                }
+            },
+            "vihMin": {
+                "description": "the least positive (most negative) value of high-level input voltage for which operation of the logic element within specification limits is to be expected",
```
do you think this wording makes more sense?

"the lowest value of high-level input voltage for which operation of the logic element within specification limits is to be expected"

asadate replied:
Yes. Changed.

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                "$ref": "../common/unit.json#/unit"
+            },
+            "vihMax": {
+                "description": "the most positive (least negative) value of high-level input voltage for which operation of the logic element within specification limits is to be expected",
```
do you think this wording makes more sense?

"the highest value of high-level input voltage for which operation of the logic element within specification limits is to be expected"

asadate replied:
Yes. Changed.

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                "$ref": "../common/unit.json#/unit"
+            },
+            "vilMax": {
+                "description": "the most positive (least negative) value of low-level input voltage for which operation of the logic element within specification limits is to be expected",
```
do you think this wording makes more sense? 

"the highest value of low-level input voltage for which operation of the logic element within specification limits is to be expected"

asadate replied:
Yes.  Changed.

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                "$ref": "../common/unit.json#/unit"
+            },
+            "vilMin": {
+                "description": "the least positive (most negative) value of low-level input voltage for which operation of the logic element within specification limits is to be expected",
```
"the lowest value of low-level input voltage for which operation of the logic element within specification limits is to be expected"

asadate replied:
Changed

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                "$ref": "../common/unit.json#/unit"
+            },
+            "inputLeakage": {
+                "description": "maximum current draw into a high impedance input pin",
```
... draw out of a high impedance input pin...

asadate replied:
Changed

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+            }
+        },
+        "pinPaths": {
+            "description": "information on pin paths - pins associated with each component in a multi-component part",
```
information on pin paths - pins associated with each other in a multi-component part (such as A1, Y1 and A2, Y2)

asadate replied:
changed

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                    "type": "string"
+                },
+                "value": {
+                    "description": "value of component if a range is not specified",
```
"value of component"

remove the part about: if a range is not specified

asadate replied:
Changed

rsnancollas replied:
this doesn't look like it was changed?

rsnancollas commented on part-spec/common/pinSpec.json:
```diff
@@ -0,0 +1,270 @@
+                    "examples": [
+                        "UART_TX",
+                        "UART_RX",
+                        "SPI_COPI",
```
Are we using COPI/CIPO or HODI/HIDO?

I believe google uses HIDO/HODI

asadate replied:
I've seen both used.  Not sure which one is more adopted today. I'll check

rsnancollas commented on part-spec/common/powerFetProperties.json:
```diff
@@ -0,0 +1,30 @@
+        "type": "object",
+        "properties": {
+            "ilimHSFET": {
+                "description": "maximum sustained current output current under which the high side FET will operate properly",
```
maximum sustained output current under which the high side FET will operate properly

asadate replied:
changed

rsnancollas commented on part-spec/common/powerFetProperties.json:
```diff
@@ -0,0 +1,30 @@
+                "$ref": "../common/unit.json#/unit"
+            },
+            "ilimLSFET": {
+                "description": "maximum sustained current output current under which the low side FET will operate properly",
```
maximum sustained output current under which the low side FET will operate properly

asadate replied:
changed.

rsnancollas commented on part-spec/common/unit.json:
```diff
@@ -0,0 +1,101 @@
+                    "megahertz",
+                    "percentage",
+                    "bit",
+                    "byte"
```
maybe add dB and ppm

asadate replied:
let's discuss.

rsnancollas replied:
we can not add these for now.

rsnancollas commented on part-spec/common/unit.json:
```diff
@@ -0,0 +1,101 @@
+                "type": "number"
+            },
+            "minValue": {
+                "description": "minimum unit quantity corresponding to unit text - example if millivolt was selected 40mV would have a value of 1, if volt was selected 40mV would have a value of 0.001",
```
I think this description is wrong:

"description": "minimum unit quantity corresponding to unit text - example if millivolt was selected 30mV would have a value of 30"

rsnancollas commented on part-spec/common/unit.json:
```diff
@@ -0,0 +1,101 @@
+                "type": "number"
+            },
+            "maxValue": {
+                "description": "maximum unit quantity corresponding to unit text - example 40mV would have a value of 40",
```
"maximum unit quantity corresponding to unit text - example 50mV would have a value of 50",

rsnancollas commented on part-spec/common/unit.json:
```diff
@@ -0,0 +1,101 @@
+                "type": "number"
+            },
+            "unitFactor": {
+                "description": "multiplier on the value to achieve the SI unit listed - example for 40mV the unitFactor would be 0.001",
```
"multiplier on the value to achieve the SI unit listed -  for example if millivolt was selected 40mV would have a value of 1, if volt was selected 40mV would have a value of 0.001"

asadate replied:
good catch. Changed.

rsnancollas commented on part-spec/common/unit.json:
```diff
@@ -0,0 +1,101 @@
+                    "subtract"
+                ]
+            },
+            "valueDefined": {
```
I question whether this is a good idea.

By default, if a unit like this is not filled in, the value will be "false", so if someone filled out a unit and just left this blank, the unit would be listed, but the "valueDefined" would be false (which is confusing and probably wrong).

I think we should maybe get rid of this.

rsnancollas commented on part-spec/component.json:
```diff
@@ -19,12 +19,12 @@
+			"$ref": "./common/package.json#/package"
 		},
 		"externalFileMap": {
 			"description": "external files that describe key component properties. External files can be used in lieu of defining core properties, pins, and package informatoin in the same file",
```
information is spelled wrong

rsnancollas commented on part-spec/data_converter/adc.json:
```diff
@@ -0,0 +1,79 @@
+            "type": "number"
+        },
+        "integralNonlinearity": {
+            "description": "deviation of an actual transfer function from an ideal transfer function, in LSB",
```
"deviation (in LSB) of an actual transfer function from an ideal transfer function",

rsnancollas commented on part-spec/data_converter/adc.json:
```diff
@@ -0,0 +1,79 @@
+        },
+        "differentialNonlinearity": {
+            "description": "difference (in LSB) in step width between the actual and ideal transfer functions",
+            "comment": "units of +/- LSB",
```
units of LSB

rsnancollas commented on part-spec/data_converter/dac.json:
```diff
@@ -0,0 +1,69 @@
+        "offsetError": {
+            "description": "analog response to an input code of all zeros",
+            "comment": "units of percentage of full scale range",
+            "type": "number"
```
should be type of units since the unit is a percentage

asadate replied:
done

rsnancollas commented on part-spec/data_converter/dac.json:
```diff
@@ -0,0 +1,69 @@
+        "gainError": {
+            "description": "difference (in percentage of FSR) of how well the slope of the actual transfer function matches the ideal transfer function",
+            "comment": "units of percentage of full scale range",
+            "type": "number"
```
type of units

asadate replied:
done

rsnancollas commented on part-spec/data_converter/dac.json:
```diff
@@ -0,0 +1,69 @@
+            ],
+            "type": "string"
+        },
+        "outputChannels": {
```
outputChannelCount

asadate replied:
done

rsnancollas commented on part-spec/hardware/connector.json:
```diff
@@ -0,0 +1,53 @@
+            "description": "property describing whether a connector has an asymmetry to prevent it from being plugged in the wrong direction",
+            "type": "boolean"
+        },
+        "numberPins": {
```
I think we can remove this - seems the same as "contactCount"

asadate replied:
Changed

rsnancollas commented on part-spec/ic_io/highspeed_mux.com:
```diff
@@ -0,0 +1,37 @@
+        "insertionLoss": {
+            "description": "insertion loss through high speed mux",
+            "comment": "units of dB",
+            "type": "number"
```
could be dB or unit depending on what we decide to support

rsnancollas replied:
sounds like we'll keep it as number

rsnancollas commented on part-spec/ic_io/level_shifter:
```diff
@@ -0,0 +1,22 @@
+    "type": "object",
+    "properties": {
+        "inputVoltage": {
+            "description": "input voltage level of level shifter",
```
nominal input voltage of level shifter

OR 

input voltage range of level shifter

asadate replied:
Done.

rsnancollas commented on part-spec/ic_io/retimer.json:
```diff
@@ -0,0 +1,63 @@
+    "type": "object",
+    "properties": {
+        "numberOfLanes": {
+            "description": "number of lanes supported by a device",
```
"number of lanes (single ended or differential) supported by retimer

asadate replied:
done

rsnancollas commented on part-spec/ic_io/retimer.json:
```diff
@@ -0,0 +1,63 @@
+            }
+        },
+        "dpMaxLinkRate": {
+            "description": "Max link rate supported by DP interface of a device",
```
maximum interface speed supported by display port interface of a device

asadate replied:
done

rsnancollas commented on part-spec/ic_io/usbc_pdcontroller.json:
```diff
@@ -0,0 +1,114 @@
+            "examples": [
+                "Source",
+                "Sink",
+                "DRP"
```
Dual Role Port

asadate replied:
done

rsnancollas commented on part-spec/ic_io/usbc_pdcontroller.json:
```diff
@@ -0,0 +1,114 @@
+            "type": "boolean"
+        },
+        "integratedLoadSwitch": {
+            "description": "whether the pd controller has one or more integrated load switches ",
```
whether the pd controller has an integrated load switch for vbus power

asadate replied:
done

rsnancollas commented on part-spec/ic_io/usbc_pdcontroller.json:
```diff
@@ -0,0 +1,114 @@
+            "description": "current used by device in various power modes",
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
```
I can't remember, was the idea of this that it would be a property for any component at the top level

asadate replied:
We can keep this at this level since this is specific for parts supporting UVLO, OVP,..

rsnancollas commented on part-spec/ic_io/usbc_pdcontroller.json:
```diff
@@ -0,0 +1,114 @@
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/ic_io/usbc_portcontroller.json:
```diff
@@ -0,0 +1,97 @@
+            "type": "string"
+        },
+        "usbTcpiRevision": {
+            "description": "usb type-c port controller interface spec revision implemented by controller",
```
I don't really understand the difference between this property and the one above it - can we combine these?

asadate replied:
These are 2 separate specifications (USB Type-C & USB TCPCI)

rsnancollas commented on part-spec/ic_io/usbc_portcontroller.json:
```diff
@@ -0,0 +1,97 @@
+            "examples": [
+                "Source",
+                "Sink",
+                "DRP"
```
Dual Role Port

asadate replied:
done

rsnancollas commented on part-spec/ic_io/usbc_portcontroller.json:
```diff
@@ -0,0 +1,97 @@
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/ic_microcontroller/microcontroller.json:
```diff
@@ -0,0 +1,60 @@
+            "description": "firmware version of the part",
+            "type": "string"
+        },
+        "activePower": {
```
I think we can remove:

activePower
standbyPower

since we have currentConsumption

asadate replied:
done

rsnancollas commented on part-spec/ic_misc/audio_codec.json:
```diff
@@ -0,0 +1,49 @@
+        "hpOutputSNR": {
+            "description": "headphone amplifier output SNR",
+            "comment": "units of dB",
+            "type": "number"
```
type: unit?

rsnancollas commented on part-spec/ic_misc/audio_codec.json:
```diff
@@ -0,0 +1,49 @@
+        "hpOutputTHD+N": {
+            "description": "headphone output total harmonic distortion plus noise",
+            "comment": "units of percentage",
+            "$ref": "number"
```
type: unit

rsnancollas replied:
this should be changed

rsnancollas commented on part-spec/ic_misc/audio_codec.json:
```diff
@@ -0,0 +1,49 @@
+        "micInputSNR": {
+            "description": "microphone input SNR",
+            "comment": "units of dB",
+            "$ref": "number"
```
type: unit?

rsnancollas commented on part-spec/ic_misc/audio_codec.json:
```diff
@@ -0,0 +1,49 @@
+        "micInputTHD+N": {
+            "description": "microphone input total harmonic distortion plus noise",
+            "comment": "units of percentage",
+            "$ref": "number"
```
type: unit

rsnancollas replied:
this should be changed

rsnancollas commented on part-spec/memory/dram.json:
```diff
@@ -0,0 +1,103 @@
+        },
+        "dataRate": {
+            "description": "dram maximum data rate",
+            "comment": "units of mega transfers per second (MT/s)",
```
should we make this a unit?

asadate replied:
Yes

rsnancollas replied:
I did you add it to "unit"?

rsnancollas reviewed (commented):
first part of the review

asadate committed:
Update clock.json

Addressed feedback from review

asadate committed:
Update oscillator.json

Addressed feedback from review

asadate committed:
Update componentProtectionThresholds.json

Addressed feedback from review.

asadate committed:
Update pinPaths.json

Addressed feedback from review

asadate committed:
Update componentID.json

Title change

asadate committed:
Update componentProtectionThresholds.json

Title change

asadate committed:
Update conditionalProperty.json

Title change.

asadate committed:
Update currentConsumption.json

Changed title.

asadate committed:
Update graph.json

Changed title & change local referenced object name from "data' to "curve".

asadate committed:
Update package.json

Changed title

asadate committed:
Update pinSpec.json

Changed title and changed local referenced object to match property name. /$def/pinFunction is now /$def/functionProperties

asadate committed:
Update register.json

Changed title, references and added a unit for the register size

asadate committed:
Update unit.json

Change title.

asadate committed:
Update powerFetProperties.json

Addressed feedback from review

asadate committed:
Update unit.json

Addressed feedback from review.

asadate committed:
Update component.json

Addressed feedback from review

asadate committed:
Update adc.json

Addressed feedback from review.

asadate committed:
Update unit.json

Addressed feedback from review.

rsnancollas commented on part-spec/ic_misc/speaker_amplifier.json:
```diff
@@ -0,0 +1,45 @@
+        "sampleRate": {
+            "description": "sample rate of data out of amplifier",
+            "comment": "units of Hz",
+            "$ref": "..common/unit.json#/unit"
```
../common/unit.json#/conditionalProperty

asadate replied:
done

rsnancollas replied:
"../common/unit.json#/unit"

rsnancollas commented on part-spec/memory/rom.json:
```diff
@@ -0,0 +1,30 @@
+            ]
+        },
+        "qeStatus": {
+            "description": "indicates whether the QE bit is set",
```
indicates whether the Quad Enable (QE) bit is set

asadate replied:
done

rsnancollas commented on part-spec/passives/capacitor.json:
```diff
@@ -0,0 +1,63 @@
+            "description": "describes whether the capacitor is polarized",
+            "type": "boolean"
+        },
+        "equivalentSerieResistance": {
```
equivalentSeriesResistance

asadate replied:
done

rsnancollas commented on part-spec/passives/capacitor.json:
```diff
@@ -0,0 +1,63 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "capacitorDerating": {
+            "description": "graph object to capture capacitance changes with voltage",
```
graph object to capture capacitance changes with voltage changes

asadate replied:
done

rsnancollas commented on part-spec/passives/inductor.json:
```diff
@@ -0,0 +1,69 @@
+    },
+    "tolerance": {
+      "description": "nominal tolerance of an inductor",
+      "comment": "units of %",
```
units of percentage

asadate replied:
done

rsnancollas commented on part-spec/passives/inductor.json:
```diff
@@ -0,0 +1,69 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "selfResonantFrequency": {
+      "description": "frequency at which the inductor becomes a capacitor",
```
frequency at which the inductor begins behaving like a capacitor

asadate replied:
done

rsnancollas commented on part-spec/passives/inductor.json:
```diff
@@ -0,0 +1,69 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "saturationCurve": {
+      "description": "graph object to capture inductor saturation with current",
```
graph object to capture inductance as a function of current

asadate replied:
done

rsnancollas commented on part-spec/passives/inductor.json:
```diff
@@ -0,0 +1,69 @@
+      "$ref": "../common/graph.json#/graph"
+    },
+    "resonantFrequencyCurve": {
+      "description": "graph object to capture inductor resonant frequency",
```
graph object to capture inductance as a function of frequency

asadate replied:
done

rsnancollas commented on part-spec/passives/resistor.json:
```diff
@@ -0,0 +1,55 @@
+    ],
+    "properties": {
+        "value": {
+            "description": "resistor value",
```
resistance value

asadate replied:
done.

rsnancollas commented on part-spec/passives/resistor.json:
```diff
@@ -0,0 +1,55 @@
+        },
+        "tolerance": {
+            "description": "nominal tolerance of a resistor",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+        },
+        "inputCurrentAccuracy": {
+            "description": "accuracy of input current when set",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+        },
+        "batteryChargeCurrentAccuracy": {
+            "description": "charging current regulation accuracy of a device",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+        },
+        "batteryPreChargeCurrentAccuracy": {
+            "description": "pre-charging current regulation accuracy of a device",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+        },
+        "batteryTrickleChargeCurrentAccuracy": {
+            "description": "Trickle charging current regulation accuracy of a device",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "batteryTrickleChargeCurrentAccuracy": {
+            "description": "Trickle charging current regulation accuracy of a device",
```
trickle

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+        },
+        "batteryTerminationChargeCurrentAccuracy": {
+            "description": "termination charging current regulation accuracy of a device",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+        },
+        "batteryChargeVoltageAccuracy": {
+            "description": "accuracy of battery charge voltage regulated by a device",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "efficiency": {
+            "description": "charge efficiency vs charge current of a device",
```
charger efficiency as a function of charge current

asadate replied:
done

rsnancollas commented on part-spec/power/battery_charger.json:
```diff
@@ -0,0 +1,218 @@
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/power/displaybacklight_driver.json:
```diff
@@ -0,0 +1,83 @@
+            "comment": "units of volts",
+            "$ref": "../common/unit.json#/unit"
+        },
+        "vout": {
```
is this fixed or should this be a range? 

I guess "unit" allows for a range, but maybe we should specify this in the description:


"output voltage range a device can regulate"

asadate replied:
changed

rsnancollas commented on part-spec/power/displaybacklight_driver.json:
```diff
@@ -0,0 +1,83 @@
+        },
+        "ioutAccuracy": {
+            "description": "accuracy of per string current regulated by a device",
+            "comment": "units of %",
```
percentage

asadate replied:
done

rsnancollas commented on part-spec/power/displaybacklight_driver.json:
```diff
@@ -0,0 +1,83 @@
+            "$ref": "../common/ratio.json#/ratio"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/power/displaybacklight_driver.json:
```diff
@@ -0,0 +1,83 @@
+            "$ref": "../common/componentProtectionThresholds.json#/componentProtectionThresholds"
+        },
+        "efficiency": {
+            "description": "efficiency vs forward current",
```
backlight driver efficiency as a function of forward current

asadate replied:
done

rsnancollas commented on part-spec/power/linear_regulator.json:
```diff
@@ -0,0 +1,88 @@
+    ],
+    "properties": {
+        "vin": {
+            "description": "input voltage under which the part can be expected to operate without the output dropping",
```
minimum input voltage under which the part can be expected to operate without the output dropping

asadate replied:
changed

rsnancollas commented on part-spec/power/linear_regulator.json:
```diff
@@ -0,0 +1,88 @@
+        "loadCurrent"
+    ],
+    "properties": {
+        "vin": {
```
vinMin

asadate replied:
This should be covered by unit.

rsnancollas commented on part-spec/power/linear_regulator.json:
```diff
@@ -0,0 +1,88 @@
+        "dropoutVoltage": {
+            "description": "dropout voltage of a device",
+            "comment": "units of volts",
+            "type": "number"
```
"$ref": "../common/unit.json#/unit"

asadate replied:
done

rsnancollas commented on part-spec/power/linear_regulator.json:
```diff
@@ -0,0 +1,88 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "powerSupplyRejectionRatio": {
+            "description": "graph object to capture Power Supply Rejection Ratio (PSRR) of device over various frequencies",
```
graph object to capture Power Supply Rejection Ratio (PSRR) of device as a function of frequency

asadate replied:
done

rsnancollas commented on part-spec/power/linear_regulator.json:
```diff
@@ -0,0 +1,88 @@
+            "$ref": "../common/graph.json#/graph"
+        },
+        "rmsOutputNoise": {
+            "description": "graph object to capture RMS output noise of device over various frequencies",
```
graph object to capture RMS output noise of device as a function of frequency

asadate replied:
done

rsnancollas commented on part-spec/power/linear_regulator.json:
```diff
@@ -0,0 +1,88 @@
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/power/load_switch.json:
```diff
@@ -0,0 +1,93 @@
+            "comment": "conditions includes vin and output current",
+            "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+        },
+        "pdResistance": {
```
pullDownResistance

asadate replied:
done

rsnancollas commented on part-spec/power/load_switch.json:
```diff
@@ -0,0 +1,93 @@
+        },
+        "fallTime": {
+            "description": "time for output voltage to go from 90% vout nominal to 10% vout nominal",
+            "comment": "conditions include output capacitor value and current output, vin",
```
conditions include output capacitor value, current output, and vin

asadate replied:
done

rsnancollas commented on part-spec/power/load_switch.json:
```diff
@@ -0,0 +1,93 @@
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+            "description": "number of analog to digital converters in the device",
+            "type": "number"
+        },
+        "externalClockCount": {
```
are these external clock and internal clocks really that useful? I might consider removing these fields

asadate replied:
Yes. Some platforms expect clocks from the PMICs and some PMICs need input clocks. No need to remove because these are optional fields.

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+            "type": "number"
+        },
+        "componentList": {
+            "description": "List of components integrated in the device",
```
list of components integrated in the device

asadate replied:
done

rsnancollas replied:
I meant "list" should be lowercase, but I can try to catch this in the script

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+        },
+        "componentList": {
+            "description": "List of components integrated in the device",
+            "comment": "list should include components with an existing digital datasheet specification",
```
should we add an example here

"example": "load switch, buck converter, ldo"

asadate replied:
done

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+            }
+        },
+        "instances": {
+            "description": "definition(specification) of each instance of a component in the device",
```
definition (specification) of each instance of a component in the device

asadate replied:
done

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+            "$ref": "../common/currentConsumption.json#/currentConsumption"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+                    ],
+                    "type": "string"
+                },
+                "instanceDefinition": {
```
I think this will work, but I'm not entirely sure you will be able to validate this with the schema now that the components are in different files

asadate replied:
Good point. Let me see if we can come up with solution similar to what we did with the top level component.

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+                        "LDO1",
+                        "BUCK1",
+                        "BUCK3",
+                        "clock1"
```
CLOCK1

asadate replied:
I purposely made this lower case when I realized all the other examples were upper cases. i didn't want to give the impression that upper case was a requirement.

rsnancollas replied:
ah I think it's kind of confusing to switch cases like this. I think if you want to make it seem like it's not case sensitive, it's better to have everything lowercase.

rsnancollas commented on part-spec/power/pmic.json:
```diff
@@ -0,0 +1,108 @@
+                "componentName": {
+                    "description": "name of the component defined in the digital datasheets specifications",
+                    "examples": [
+                        "linear_regulator",
```
don't use _ 

so 
linear regulator 
switching regulator
load switch
display backlight driver
adc
clock

asadate replied:
done

rsnancollas commented on part-spec/power/switching_regulator.json:
```diff
@@ -0,0 +1,95 @@
+            "type": "string"
+        },
+        "vin": {
+            "description": "input voltage under which a device can be expected to operate without the output dropping",
```
input voltage range under which a device can be expected to reliably operate

asadate replied:
done

rsnancollas commented on part-spec/power/switching_regulator.json:
```diff
@@ -0,0 +1,95 @@
+            "$ref": "../common/powerFetProperties.json#/powerFetProperties"
+        },
+        "componentProtectionThresholds": {
+            "description": "Thermal and power supply protection thresholds of a device",
```
thermal

asadate replied:
done

rsnancollas commented on part-spec/power/switching_regulator.json:
```diff
@@ -0,0 +1,95 @@
+            "$ref": "../common/componentProtectionThresholds.json#/componentProtectionThresholds"
+        },
+        "efficiency": {
+            "description": "power efficiency of regulator",
```
power efficiency of a regulator as a function of output current?

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/bjt.json:
```diff
@@ -0,0 +1,150 @@
+        },
+        "emitterBaseCutOffCurrent": {
+            "description": "current into the base terminal when the BJT's base and emitter are reverse biased and the collector is open (I_EBO)",
+            "$comment": "conditions include emitter-base Voltage",
```
voltage

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/bjt.json:
```diff
@@ -0,0 +1,150 @@
+            "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+        },
+        "icVsHfe": {
+            "description": "graph of collector current (Ic) vs dc current gain (Hfe).",
```
graph of collector current (Ic) as a function of dc current gain (Hfe)

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/bjt.json:
```diff
@@ -0,0 +1,150 @@
+            "$ref": "../common/graph.json#/graph"
+        },
+        "icVsVce": {
+            "description": "graph of collector current (Ic) vs collector-emitter saturation voltage (VCE_SAT).",
```
vgraph of collector current (Ic) as a function of collector-emitter saturation voltage (VCE_SAT)

rsnancollas commented on part-spec/semiconductor/bjt.json:
```diff
@@ -0,0 +1,150 @@
+            "$ref": "../common/graph.json#/graph"
+        },
+        "ibVsVce": {
+            "description": "graph of base current (Ib) vs collector-emitter voltage (VCE).",
```
graph of base current (Ib) as a function of collector-emitter voltage (VCE)

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/bjt.json:
```diff
@@ -0,0 +1,150 @@
+            "$ref": "../common/graph.json#/graph"
+        },
+        "pdVsTemp": {
+            "description": "graph of power dissipation vs temperature.",
```
graph of power dissipation as a function of temperature

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/diode.json:
```diff
@@ -0,0 +1,115 @@
+            "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+        },
+        "ifVsVf": {
+            "description": "graph of forward current (If) vs forward voltage (VfTyp)",
```
graph of forward current (If) as a function of forward voltage (VfTyp)

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/led.json:
```diff
@@ -0,0 +1,72 @@
+        },
+        "ledCapacitance": {
+            "description": "capacitance of an LED",
+            "$comment": "units of pico Farads (pF)",
```
units of farads

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/led.json:
```diff
@@ -0,0 +1,72 @@
+            "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+        },
+        "dominantWavelength": {
+            "description": " dominant wavelength an LED emits the majority of the time",
```
remove space before dominant

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/led.json:
```diff
@@ -0,0 +1,72 @@
+        },
+        "pd": {
+            "description": "power dissipation of an LED",
+            "$comment": "units of milliwatts (mW)",
```
units of watts

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/led.json:
```diff
@@ -0,0 +1,72 @@
+            "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+        },
+        "angleHalfIntensity": {
+            "description": " angle at which LED intensity falls to 50% of its maximum value",
```
remove space before angle

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "vgsMax": {
+      "description": "maximum gate to source voltage difference that can be continously applied to a MOSFET. This is a limiting value.",
```
maximum gate to source voltage difference that can be continuously applied to a MOSFET

asadate replied:
This is an input from Nexperia. The wanted us to add this.

rsnancollas replied:
I don't know what that means though, but I guess it's okay

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "vdsMax": {
+      "description": "maximum drain to source voltage difference that can be continously applied to a MOSFET. This is a limiting value.",
```
maximum drain to source voltage difference that can be continuously applied to a MOSFET

asadate replied:
Same as above.

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "iD": {
+      "description": "Drain Current of a MOSFET.",
```
drain current of a MOSFET

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "iDrain": {
+      "description": "maximum continous DC current that can flow through a MOSFET channel.This is a limiting value.",
```
maximum continuous DC current that can flow through a MOSFET channel

asadate replied:
same as above

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "idPulsed": {
+      "description": "maximum pulsed DC current that can flow through a MOSFET channel.This is a limiting value.",
```
maximum pulsed DC current that can flow through a MOSFET channel

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+    },
+    "diodeContinuousCurrent": {
+      "description": "maximum continuous forward current of the body diode of a MOSFET (IS).This is a limiting value.",
```
maximum continuous forward current of the body diode of a MOSFET (IS)

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/unit.json#/unit"
+    },
+    "diodePulsedCurrent": {
+      "description": "maximum pulsed forward current of the body diode of a MOSFET. This is a limiting value.",
```
maximum pulsed forward current of the body diode of a MOSFET

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/conditionalProperty.json#/conditionalProperty"
+    },
+    "idVsVds": {
+      "description": "graph of drain current (iD) vs drain source voltage (vds).",
```
graph object of drain current (iD) as a function of drain source voltage (vds)

asadate replied:
done

rsnancollas commented on part-spec/semiconductor/mosfet.json:
```diff
@@ -0,0 +1,195 @@
+      "$ref": "../common/graph.json#/graph"
+    },
+    "idVsVgs": {
+      "description": "graph of drain current (iD) vs gate source voltage (vgs).",
```
graph object of drain current (iD) as a function of gate source voltage (vgs)

asadate replied:
done

rsnancollas commented on part-spec/sensor/accelerometer.json:
```diff
@@ -0,0 +1,108 @@
+                "2000"
+            ],
+            "comment": "units of LSB/g (least significant bits per g) at a given acceleration range and output resolution",
+            "$ref": "../common/unit.json#/unit"
```
this should probably be a number, unless we want to support one-off units in "unit"

asadate replied:
done

rsnancollas commented on part-spec/sensor/accelerometer.json:
```diff
@@ -0,0 +1,108 @@
+        "accelerationSensitivityOverTemperature": {
+            "description": "accelerometer sensitivity change over temperature",
+            "comment": "units of percentage per degree celsius (%/Celsius)",
+            "$ref": "../common/unit.json#/unit"
```
same comment, should this be number

asadate replied:
done

rsnancollas commented on part-spec/sensor/accelerometer.json:
```diff
@@ -0,0 +1,108 @@
+        "zerogOffset": {
+            "description": "output of the accelerometer when no acceleration is applied",
+            "comment": "units of g-force (g)",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/accelerometer.json:
```diff
@@ -0,0 +1,108 @@
+        "zerogOffsetOverTemperature": {
+            "description": "accelerometer zero-g offset change over temperature",
+            "comment": "units of milli g-force per degree celsius (mg/Celcius)",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/accelerometer.json:
```diff
@@ -0,0 +1,108 @@
+        "rmsNoise": {
+            "description": "broadband rms noise of a device",
+            "comment": "units of milli g-force",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/accelerometer.json:
```diff
@@ -0,0 +1,108 @@
+        "spectralNoiseDensity": {
+            "description": "spectral noise density of a device",
+            "comment": "units of milli g-force per square root of Hz (milli-g/rt-Hz)",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/gyroscope.json:
```diff
@@ -0,0 +1,81 @@
+        },
+        "gyroSensitivityOverTemperature": {
+            "description": "gyroscope sensitivity change over temperature",
+            "comment": "units of percentage per degree celsius",
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/gyroscope.json:
```diff
@@ -0,0 +1,81 @@
+        "zeroRateOffset": {
+            "description": "output of the gyroscope when no angular velocity is applied",
+            "comment": "units of degrees-per-second",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/gyroscope.json:
```diff
@@ -0,0 +1,81 @@
+        "zeroRateOffsetOverTemperature": {
+            "description": "gyro zero rate offset change over temperature",
+            "comment": "units of degrees-per-second per degree celsius (dps/Celsius)",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/gyroscope.json:
```diff
@@ -0,0 +1,81 @@
+        "rmsNoise": {
+            "description": "broadband rms noise of a device",
+            "comment": "units of degrees-per-second rms",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/gyroscope.json:
```diff
@@ -0,0 +1,81 @@
+        "spectralNoiseDensity": {
+            "description": "spectral noise density of a device",
+            "comment": "units of degrees-per-second per square root Hertz (dps/rt-Hz)",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/magnetic_sensor.json:
```diff
@@ -0,0 +1,76 @@
+        "sensitivity": {
+            "description": "this is the gain - amount of change in the output voltage for a change in the magnetic flux density",
+            "comment": "units of mV/mT",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/magnetic_sensor.json:
```diff
@@ -0,0 +1,76 @@
+        "linearMagneticSensingRange": {
+            "description": "magnetic flux density range over which the output voltage is linear ",
+            "comment": "units of mT",
+            "$ref": "../common/unit.json#/unit"
```
unit or number?

asadate replied:
number

rsnancollas commented on part-spec/sensor/magnetic_sensor.json:
```diff
@@ -0,0 +1,76 @@
+        "operatePoint": {
+            "description": "magnetic flux density threshold which causes the sensor output to turn on",
+            "comment": "units of mT. Depending on the sensor's design, the operation point can cause the output to trip High or Low.",
+            "$ref": "../common/unit.json#/unit"
```
number, not unit

asadate replied:
changed

rsnancollas commented on part-spec/sensor/magnetic_sensor.json:
```diff
@@ -0,0 +1,76 @@
+        "releasePoint": {
+            "description": "magnetic flux density threshold which causes the sensor output to turn off",
+            "comment": "units of mT. Depending on the sensor's design, the release point can cause the output to trip High or Low.",
+            "$ref": "../common/unit.json#/unit"
```
number, not unit

asadate replied:
number

rsnancollas commented on part-spec/sensor/magnetic_sensor.json:
```diff
@@ -0,0 +1,76 @@
+        "hysteresis": {
+            "description": "delta between the operate point and the release point threshold",
+            "comment": "units of mT",
+            "$ref": "../common/unit.json#/unit"
```
number, not unit

asadate replied:
changed

rsnancollas commented on part-spec/sensor/thermal_sensor.json:
```diff
@@ -0,0 +1,61 @@
+        "gain": {
+            "description": "amount of change in the output voltage for a change in temperature",
+            "comment": "units of mV/degrees C",
+            "$ref": "../common/unit.json#/unit"
```
number

rsnancollas commented on part-spec/storage/sd_card.json:
```diff
@@ -0,0 +1,35 @@
+        "dataRate": {
+            "description": "maximum data rate of teh sd card",
+            "comment": "units of bytes per second (Bps)",
+            "$ref": "../common/unit.json#/unit"
```
number

rsnancollas commented on part-spec/storage/sd_card.json:
```diff
@@ -0,0 +1,35 @@
+            "$ref": "../common/unit.json#/unit"
+        },
+        "dataRate": {
+            "description": "maximum data rate of teh sd card",
```
maximum data rate of the sd card

rsnancollas commented on part-spec/storage/ssd.json:
```diff
@@ -0,0 +1,43 @@
+        "dataRate": {
+            "description": "maximum data rate of the ssd",
+            "comment": "units of bytes per second (Bps)",
+            "$ref": "../common/unit.json#/unit"
```
number

rsnancollas reviewed (commented):
finished the review :).

asadate committed:
Update dac.json

Addressed Changes from review

asadate committed:
Update connector.json

Addressed changes from the review

asadate committed:
Update connector.json

asadate committed:
Update level_shifter

Addressed feedback from review.

asadate committed:
Update retimer.json

Addressed feedback from review.

asadate committed:
Update usbc_pdcontroller.json

Addressed changes from review.

asadate committed:
Update usbc_portcontroller.json

Addressed feedback from review

asadate committed:
Update microcontroller.json

Addressed feedback from review

asadate committed:
Update speaker_amplifier.json

Addressed feedback from review

asadate committed:
Update dram.json

Addressed feedback from review

asadate committed:
Update rom.json

Addressed feedback from review

asadate committed:
Update capacitor.json

Addressed feedback from review

asadate committed:
Update inductor.json

Addressed feedback from review

asadate committed:
Update resistor.json

Addressed feedback from review

asadate committed:
Update battery_charger.json

Addressed feedback from review

asadate committed:
Update displaybacklight_driver.json

Addressed feedback from review

asadate committed:
Update linear_regulator.json

Addressed feedback from review

asadate committed:
Update load_switch.json

Addressed feedback from review

asadate committed:
Update pinSpec.json

Addressed feedback from review

asadate committed:
Update pmic.json

Addressed feedback from review

asadate committed:
Update switching_regulator.json

Addressed feedback from review

asadate committed:
Update bjt.json

Addressed feedback from review

asadate committed:
Update diode.json

Addressed changes from feedback

asadate committed:
Update led.json

Addressed feedback from review

asadate committed:
Update mosfet.json

Addressed feedback from review

asadate committed:
Update accelerometer.json

Addressed feedback from review

asadate committed:
Update gyroscope.json

Addressed feedback from review

asadate committed:
Update magnetic_sensor.json

Addressed feedback from review

asadate committed:
Update thermal_sensor.json

Addressed feedback from review

asadate committed:
Update sd_card.json

Addressed feedback from review

asadate committed:
Update ssd.json

Addressed feedback from review

rsnancollas reviewed (commented):
I'm going to approve and merge this so that I can start making changes. There are a few things that didn't get changed, but I'll put up a smaller review to change those since this really huge PR is kind of hard to review.

rsnancollas reviewed (approved):
approving to get merged in
